cogs/helpermessage.py

```diff
@@ -0,0 +1,149 @@
+        if before.roles != after.roles:
+            before_roles = set(before.roles)
+            after_roles = set(after.roles)
+            # checking whether roles were removed or added
+            added_roles = after_roles - before_roles
+            removed_roles = before_roles - after_roles
+            # routine that runs to add the role
+            if added_roles:
+                async for helpermessage, role, embed in self.embed_getter(added_roles):
+                    new_embed = discord.Embed(description=embed.description)
+                    for field in embed.fields:
+                        if field.name == f'**{role.name}**':
+                            new_embed.add_field(name=field.name, value=field.value + f'\n{after.mention}')
+                        else:
+                            new_embed.add_field(name=field.name, value=field.value)
+                    await helpermessage.edit(embed=new_embed)
+
+            # routine that runs to remove roles
+            if removed_roles:
+                async for helpermessage, role, embed in self.embed_getter(removed_roles):
+                    new_embed = discord.Embed(description=embed.description)
+                    for field in embed.fields:
+                        if field.name == f'**{role.name}**':
+                            members = field.value.split('\n')
+                            new_member_list = [member for member in members if member != after.mention]
+                            new_embed.add_field(name=field.name, value='\n'.join(new_member_list))
+                        else:
+                            new_embed.add_field(name=field.name, value=field.value)
+                    await helpermessage.edit(embed=new_embed)
+
+    @helpermessage.command()
+    async def create(self, ctx: commands.Context, channel: discord.TextChannel, helper_roles: commands.Greedy[discord.Role]):
+        """
+        Create the helpermessage embed for a specific channel given the channel and helper role
+        """
+        # Create the embed that will be pinned in the message
+        description = self.description['description'] + f"\n\n **Subject helpers for {channel.name}:**"
+        helper_embed = discord.Embed(description=description)
+        # add the field displaying the helpers
+        for role in helper_roles:
+            subject_helpers = [member.mention for member in ctx.guild.members if role in member.roles]
+            helper_embed.add_field(
+                name=f"**{role.name}**",
+                value="\n".join(subject_helpers),
+                inline=False
+            )
+        # send the message
+        helpermessage = await channel.send(embed=helper_embed)
+        # pin the message
+        await helpermessage.pin()
+        # save to the database
+        values = dict(
+            message_id = helpermessage.id,
+            channel_id = channel.id,
+            role_id = [role.id for role in helper_roles]
```

Comment:
Since role_id is an array, the field in `db/models.py` should be updated to `ArrayField()`